lib/SILOptimizer/Utils/Local.cpp

```diff
@@ -208,6 +208,10 @@ void swift::eraseUsesOfValue(SILValue V) {
     auto *User = UI->getUser();
     UI++;
 
+    // User could be removed at this point.
```

Comment:
Is this the right fix? If the User is removed are we guaranteed to have it be nulled out?

Why not use a SmallPointerSet.